### PR TITLE
AMC Banking Fundamentals 365 V21.4 HF1

### PR DESCRIPTION
**Pull request by AMCJN**

### CODE REVIEW

File: Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankServiceRequestMgt.Codeunit.al.bak [hunk 1]
```diff
@@ -0,0 +1,872 @@
+codeunit 20118 "AMC Bank Service Request Mgt."
```
Comment (haoranpb) [inline Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankServiceRequestMgt.Codeunit.al.bak:1]: @mazhelez should I add this to gitignore?

File: Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankAssistedMgt.Codeunit.al [hunk 2]
```diff
@@ -62,6 +62,7 @@ codeunit 20117 "AMC Bank Assisted Mgt."
         AMCBankImpBankListHndl: Codeunit "AMC Bank Imp.BankList Hndl";
         LongTimeout: Integer;
         ShortTimeout: Integer;
+        AMCBoughtModule: Boolean; //Hotfix AMC-JN
```
Comment (haoranpb) [inline Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankAssistedMgt.Codeunit.al:65]:
```suggestion
        AMCBoughtModule: Boolean;
```
Notice lots of the same comment in the PR, is it really useful?

File: Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankingMgt.Codeunit.al [hunk 3]
```diff
@@ -623,5 +660,12 @@ codeunit 20105 "AMC Banking Mgt."
         else
             exit(BankAccount."AMC Bank Name" + FileExtTxt);
     end;
+
+    [NonDebuggable]
+    internal procedure AMCTenantId(): Text //Hotfix AMC-JN
+    var
+    begin
+        Exit(CopyStr('fb79e895-7de3-4468-8184-cd181eb8b131', 1, 40)); //AMC Tenant Guid
```
Comment (haoranpb) [inline Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankingMgt.Codeunit.al:668]:
```suggestion
        Exit(CopyStr('fb79e895-7de3-4468-8184-cd181eb8b131', 1, 40));
```
Similar here, the procedure name already reflects what's the purpose

File: Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankingMgt.Codeunit.al [hunk 4]
```diff
@@ -623,5 +660,12 @@ codeunit 20105 "AMC Banking Mgt."
         else
             exit(BankAccount."AMC Bank Name" + FileExtTxt);
     end;
+
+    [NonDebuggable]
```
Comment (haoranpb) [inline Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankingMgt.Codeunit.al:664]: The code is open sourced, not sure if `NonDebuggable`  would be very helpful

File: Apps/W1/AMCBanking365Fundamentals/test/src/Bank/AMCBankingCredentialUT.Codeunit.al [hunk 5]
```diff
@@ -15,7 +15,8 @@ codeunit 132558 "AMC Banking Credential UT"
         LibraryXPathXMLReader: Codeunit "Library - XPath XML Reader";
         EnvironmentInfoTestLibrary: Codeunit "Environment Info Test Library";
         AMCBankingMgt: Codeunit "AMC Banking Mgt.";
-        LocalhostURLTxt: Label 'https://localhost:8080/', Locked = true;
+        //LocalhostURLTxt: Label 'https://localhost:8080/', Locked = true;
+        LocalhostURLTxt: Label 'https://host.docker.internal:8088/', Locked = true; //TODO - Put back to above when finished testing
```
Comment (haoranpb) [inline Apps/W1/AMCBanking365Fundamentals/test/src/Bank/AMCBankingCredentialUT.Codeunit.al:19]: TODO?

File: Apps/W1/AMCBanking365Fundamentals/test/src/Libraries/LibraryAMCWebService.Codeunit.al [hunk 6]
```diff
@@ -7,7 +7,9 @@ codeunit 130103 "Library - Amc Web Service"
 
     var
         AMCBankingMgt: Codeunit "AMC Banking Mgt.";
-        LocalhostURLTxt: Label 'https://localhost:8080/', Locked = true;
+        //LocalhostURLTxt: Label 'https://localhost:8080/', Locked = true;
+        LocalhostURLTxt: Label 'https://host.docker.internal:8088/', Locked = true; //TODO - Put back to above when finished testing
```
Comment (haoranpb) [inline Apps/W1/AMCBanking365Fundamentals/test/src/Libraries/LibraryAMCWebService.Codeunit.al:11]: TODO?

File: Apps/W1/AMCBanking365Fundamentals/test/src/Libraries/LibraryPaymentAMC.Codeunit.al [hunk 7]
```diff
@@ -6,7 +6,8 @@ codeunit 130102 "Library - Payment AMC"
     end;
 
     var
-        LocalhostURLTxt: Label 'https://localhost:8080/', Locked = true;
+        //    LocalhostURLTxt: Label 'https://localhost:8080/', Locked = true;
+        LocalhostURLTxt: Label 'https://host.docker.internal:8088/', Locked = true; //TODO - Put back to above when finished testing
```
Comment (haoranpb) [inline Apps/W1/AMCBanking365Fundamentals/test/src/Libraries/LibraryPaymentAMC.Codeunit.al:10]: TODO

**Comment by haoranpb** [comment]
a general question, which branch do you expect this PR to go in?

Is master branch enough? Or it needs to be hotfixed into 21.4?

**Review by haoranpb — CHANGES_REQUESTED** [review]

**Comment by AMCJN** [comment]
> a general question, which branch do you expect this PR to go in?
> 
> Is master branch enough? Or it needs to be hotfixed into 21.4?

Just to master branch.

File: Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankingMgt.Codeunit.al (hunk 4 above)
Comment (AMCJN) [reply Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankingMgt.Codeunit.al:664]: Just remove the NonDebuggable

File: Apps/W1/AMCBanking365Fundamentals/test/src/Libraries/LibraryPaymentAMC.Codeunit.al (hunk 7 above)
Comment (haoranpb) [reply Apps/W1/AMCBanking365Fundamentals/test/src/Libraries/LibraryPaymentAMC.Codeunit.al:10]: I'm not sure why is this resolved, is this PR ready? a bit confused tbh

Comment (AMCJN) [reply Apps/W1/AMCBanking365Fundamentals/test/src/Libraries/LibraryPaymentAMC.Codeunit.al]: Hi Sun Haoran,

Sorry for the confusing.

The line no. 9 is the correct line,
Line no. 10 should be outcommented //

I have have changed it in my respository  https://github.com/microsoft/ALAppExtensions/commit/d245f9ed983a734af894b8faff592c4eec484f66

Can you fix this in this pull request or must I cancel this and make a new one?

Br
AMC /JN

Comment (haoranpb) [reply Apps/W1/AMCBanking365Fundamentals/test/src/Libraries/LibraryPaymentAMC.Codeunit.al]: I can see the changes now, porting this internally at the moment, will post update here

File: Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankAssistedMgt.Codeunit.al [hunk 8]
```diff
@@ -80,6 +81,7 @@ codeunit 20117 "AMC Bank Assisted Mgt."
         end;
 
         if ((AMCBankingSetup."User Name" <> AMCBankingSetup.GetDemoUserName()) and
+            (not AMCBankingMgt.IsLicenseEqualAMC) and //Hotfix AMC-JN
```
Comment (haoranpb) [inline Apps/W1/AMCBanking365Fundamentals/app/Codeunits/AMCBankAssistedMgt.Codeunit.al:84]:
```suggestion
            (not AMCBankingMgt.IsLicenseEqualAMC()) and //Hotfix AMC-JN
```

Small changes like this are fixed internally

**Review by haoranpb — APPROVED** [review]

**Comment by haoranpb** [comment]
@AMCJN PR is merging into master at the moment.

**Comment by haoranpb** [comment]
@AMCJN some colleagues pointed out that we have lots of comments in code that looks un-used, can we remove them when merge?

e.g.

```
//Hotfix AMC-JN
```

**Comment by AMCJN** [comment]
> @AMCJN some colleagues pointed out that we have lots of comments in code that looks un-used, can we remove them when merge?
> 
> e.g.
> 
> ```
> //Hotfix AMC-JN
> ```

Yes, please remove. :-)

**Comment by haoranpb** [comment]
PR merged internally in master

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 53998.

**Pull request closed without merging**